cli/azd/pkg/infra/provisioning/manager.go

```diff
@@ -31,7 +31,21 @@ type Manager struct {
 	options             *Options
 }
 
+// defaultOptions for this package. These values are applied across provisioning providers.
+var defaults = Options{
+	Module: "main",
+	Path:   "infra",
+}
+
 func (m *Manager) Initialize(ctx context.Context, projectPath string, options Options) error {
+	// applied defaults if missing
+	if options.Module == "" {
+		options.Module = defaults.Module
+	}
+	if options.Path == "" {
+		options.Path = defaults.Path
+	}
```

Comment:
We're losing the ability to have provider specific defaults now by removing these checks from the provider and into the generic manager struct.

Reply:
Agreed that we don't lose any ability here... The code is closer to the current world, we can always redefine it to not be infra specific if we find the need to.